src/App.tsx

```diff
@@ -1,26 +1,316 @@
+import {
+  USER_ID,
+  createTodo,
+  deleteTodos,
+  getTodos,
+  toggleUpdateTodo,
+  updateTodo,
+} from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { TodosFilter } from './components/TodosFilter/TodosFilter';
+import { FilterStatus } from './types/FilterStatus';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [title, setTitle] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState<boolean | string>(false);
+  const [disabledInput, setDisabledInput] = useState(false);
+  const [filterStatus, setFilterStatus] = useState<FilterStatus>(
+    FilterStatus.all,
+  );
+
+  const ERROR_ADD_TODO = 'Unable to add a todo';
+  const ERROR_LOAD_TODO = 'Unable to load todos';
+  const ERROR_UPDATE_TODO = 'Unable to update a todo';
+  const ERROR_DELETE_TODO = 'Unable to delete a todo';
+  const ERROR_EMPTY_INPUT = 'Title should not be empty';
+
+  const errorMessageAddTodo = errorMessage === ERROR_ADD_TODO;
+
+  const allCompleted = todos.every(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+  const todosLength = todos.length;
+
+  const resetError = () => {
+    setTimeout(() => setErrorMessage(''), 3000);
+  };
+
+  useEffect(() => {
+    setErrorMessage('');
+
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage(ERROR_LOAD_TODO))
+      .finally(resetError);
+  }, []);
+
+  const todoField = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (todoField.current) {
+      todoField.current.focus();
+    }
+  }, [todosLength, errorMessageAddTodo]);
+
+  const handleFilterChange = (status: FilterStatus) => {
+    setFilterStatus(status);
+  };
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filterStatus) {
+      case 'active':
+        return !todo.completed;
+
+      case 'completed':
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
+
+  const addTodo = async (newTitle: string) => {
+    setTempTodo({ id: 0, title: newTitle, completed: false, userId: USER_ID });
+    setIsLoading('adding');
+    setDisabledInput(true);
+
+    return createTodo({
+      title: newTitle.trim(),
+      completed: false,
+      userId: USER_ID,
+    })
+      .then(newTodo => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+        setTempTodo(null);
+      })
+      .catch(error => {
+        setTempTodo(null);
+
+        if (error instanceof Error) {
+          throw new Error(ERROR_ADD_TODO);
+        }
+
+        setErrorMessage((error as Error).message);
+        throw error;
```

Comment:
you've already handled this error - no need to throw it again - FYI

It could be a case if you handle only part of errors, like this `if (error instanceof Error)`. Usually, it's some specific type of error, for example:
```
class ValidationError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'ValidationError';
  }
}
```
I think that was your idea, but you simplified it and didn't create a separate class. That's nice you're adding error handling :+1: 

But here you did it, then **handled all other errors** and the throw. So throwing seems redundant - FYI